bin/maintenance/update_header.py

```diff
@@ -124,41 +127,26 @@ def _update_header(project, file_path, year, substring, regex, data):
         with open(file_path, 'wb') as file_write:
             file_write.write(content)
     elif not found:
-        print(cformat('%{yellow}Missing header in %{yellow!}{}').format(os.path.relpath(file_path)))
+        print('Missing header in {}'.format(os.path.relpath(file_path)))
+        return True
 
 
 def update_header(project, file_path, year):
     ext = file_path.rsplit('.', 1)[-1]
     if ext not in SUPPORTED_FILES or not os.path.isfile(file_path):
-        return
+        return False
     if os.path.basename(file_path)[0] == '.':
-        return
-    _update_header(project, file_path, year, SUBSTRING, SUPPORTED_FILES[ext]['regex'], SUPPORTED_FILES[ext]['format'])
+        return False
+    return _update_header(project, file_path, year, SUBSTRING, SUPPORTED_FILES[ext]['regex'],
+                          SUPPORTED_FILES[ext]['format'])
 
 
-def _process_args(args):
+def _process_year(ctx, param, value):
     year_regex = re.compile('^[12][0-9]{3}$')  # Year range 1000 - 2999
```

Comment:
i don't think we need to keep this. it was just to be "smart" to have optional positional arguments that aren't at the end

Reply:
But isn't it intended to also check if you provided a year in the right format instead of, say, `foo`?

Reply:
I'd say requiring a numeric value is fine for that... it's not like this tool is exposed to untrusted users. OR a simple `2020 <= year <= 2999` (as if indico (or humanity) will still be a thing by then :D)

Reply:
there's actually a click validator for this: https://click.palletsprojects.com/en/7.x/api/#click.IntRange